src/wolff_producer.erl

```diff
@@ -712,18 +766,24 @@ handle_overflow(St, Overflow) when Overflow =< 0 ->
     St;
 handle_overflow(#{replayq := Q,
                   pending_acks := PendingAcks,
-                  config := Config
+                  config := Config = #{max_batch_bytes := BytesLimit}
                  } = St,
                 Overflow) ->
   {NewQ, QAckRef, Items} =
     replayq:pop(Q, #{bytes_limit => Overflow, count_limit => 999999999}),
   ok = replayq:ack(NewQ, QAckRef),
   {FlatBatch, Calls} = get_flat_batch(Items, [], []),
   [_ | _] = FlatBatch, %% assert
-  FlatBatchSize = length(FlatBatch),
-  wolff_metrics:dropped_queue_full_inc(Config, FlatBatchSize),
-  wolff_metrics:dropped_inc(Config, FlatBatchSize),
-  wolff_metrics:queuing_change(Config, -FlatBatchSize),
+  %% since we're discarding a number of messages which would become
+  %% batches, the best we can do is estimate to how many batches those
+  %% would correspond to.
+  EstimatedNumOfBatches = case BytesLimit == 0 of
+                              true -> 1;
+                              false -> round(math:ceil(Overflow / BytesLimit))
+                          end,
```

Comment:
the number of batches is `length(Items)`.

Reply:
which equals to `length(Calls)`

Reply:
:see_no_evil: 

:heavy_check_mark: